File: test-review-visual.js
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch({ headless: false });
  const context = await browser.newContext({
    viewport: { width: 1280, height: 720 }
  });
  const page = await context.newPage();

  try {
    console.log('1. Navigating to Dallas page...');
    await page.goto('http://localhost:4321/tx/dallas');
    await page.waitForTimeout(2000);

    console.log('2. Clicking on Perot Museum venue...');
    const venueLink = page.locator('a[href*="perot-museum"]').first();
    await venueLink.click();
    await page.waitForTimeout(2000);

    console.log('3. Current URL:', page.url());

    // Take screenshot of venue page
    await page.screenshot({ path: 'screenshot-1-venue-page.png', fullPage: true });
    console.log('   Screenshot saved: screenshot-1-venue-page.png');

    console.log('4. Looking for Write a review button...');
    const reviewButton = page.locator('text=Write a review');
    const buttonCount = await reviewButton.count();
    console.log('   Found', buttonCount, 'review buttons');

    if (buttonCount > 0) {
      const href = await reviewButton.getAttribute('href');
      console.log('   Review button href:', href);

      console.log('5. Clicking Write a review button...');
      await reviewButton.click();
      await page.waitForTimeout(2000);

      console.log('6. After click URL:', page.url());

      // Take screenshot of login page
      await page.screenshot({ path: 'screenshot-2-login-page.png', fullPage: true });
      console.log('   Screenshot saved: screenshot-2-login-page.png');

      // Check what's on the page
      const pageTitle = await page.title();
      const h1Text = await page.locator('h1').first().textContent();
      console.log('   Page title:', pageTitle);
      console.log('   H1 text:', h1Text);

      // Look for email input
      const emailInput = page.locator('input[type="email"]');
      const emailCount = await emailInput.count();
      console.log('   Email inputs found:', emailCount);

      if (emailCount > 0) {
        console.log('✅ SUCCESS: Login page loaded correctly!');
        console.log('   The review button now works and redirects to the login page.');
      } else {
        console.log('❌ ISSUE: Email input not found on the page');
      }
    } else {
      console.log('❌ No review button found on the page');
    }

    await page.waitForTimeout(3000);
  } catch (error) {
    console.error('Error:', error.message);
    await page.screenshot({ path: 'screenshot-error.png', fullPage: true });
  } finally {
    await browser.close();
  }
})();

File: test-review-visual.py
#!/usr/bin/env python3
from playwright.sync_api import sync_playwright
import time

def test_review_button():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=False)
        page = browser.new_page(viewport={'width': 1280, 'height': 720})

        try:
            print("1. Navigating to venue page...")
            page.goto('http://localhost:4321/venue/perot-museum-of-nature-and-science')
            time.sleep(2)

            # Take screenshot of venue page
            page.screenshot(path='screenshot-1-venue-page.png', full_page=True)
            print("   Screenshot saved: screenshot-1-venue-page.png")

            print("2. Current URL:", page.url)

            print("3. Looking for 'Write a review' button...")
            review_buttons = page.locator('text=Write a review')
            button_count = review_buttons.count()
            print(f"   Found {button_count} review button(s)")

            if button_count > 0:
                href = review_buttons.get_attribute('href')
                print(f"   Review button href: {href}")

                print("4. Clicking 'Write a review' button...")
                review_buttons.click()
                time.sleep(2)

                print("5. After click URL:", page.url)

                # Take screenshot of login page
                page.screenshot(path='screenshot-2-after-click.png', full_page=True)
                print("   Screenshot saved: screenshot-2-after-click.png")

                # Check what's on the page
                page_title = page.title()
                h1_text = page.locator('h1').first.text_content()
                print(f"   Page title: {page_title}")
                print(f"   H1 text: {h1_text}")

                # Look for email input
                email_inputs = page.locator('input[type="email"]')
                email_count = email_inputs.count()
                print(f"   Email inputs found: {email_count}")

                if email_count > 0 and 'login' in page.url.lower():
                    print("\n✅ SUCCESS: Review button works correctly!")
                    print("   - Redirected to login page")
                    print("   - Login form is displayed")
                else:
                    print("\n⚠️  Unexpected result - check screenshots")
            else:
                print("\n❌ No review button found on the page")

            time.sleep(3)

        except Exception as e:
            print(f"\n❌ Error: {e}")
            page.screenshot(path='screenshot-error.png', full_page=True)
        finally:
            browser.close()

if __name__ == '__main__':
    test_review_button()
